module-cellular/modem/ATCommon.cpp

```diff
@@ -156,3 +162,108 @@ Result Channel::checkResult(bsp::cellular::CellularResultCode cellularResult)
 
     return result;
 }
+
+std::string Channel::anonymizeIfNecessary(std::string text)
+{
+    std::string substringCPIN = "CPIN";
+    std::string substringCLCC = "CLCC";
+    std::string substringATD  = "ATD";
+    std::string substringCPBR = "CPBR";
+    std::string substringCSPN = "CSPN";
+
+    if (text.find(substringCPIN) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text);
+    }
+    else if (text.find(substringCLCC) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 2);
+    }
+    else if (text.find(substringATD) != std::string::npos) {
+        text = anonymizeNumbers(text, 2);
+    }
+    else if (text.find(substringCPBR) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 1);
+    }
+    else if (text.find(substringCSPN) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 2);
+    }
+
+    return text;
+}
+std::string Channel::anonymizeInQuotationMarks(std::string textToAnonymize, size_t singsToLeft)
+{
+    size_t startPosition = 0;
+    size_t endPosition   = 0;
+
+    // Find the position of the substring within the input string
+    while (startPosition < textToAnonymize.length()) {
+        startPosition = textToAnonymize.find('\"', startPosition);
```

Comment:
I'd move this token to some `constexpr`, as it might not be obvious what it represents